### PR TITLE
ROX-25048: S3 compatible integration forms

### PR DESCRIPTION
**Pull request by stehessel**
### Description

This adds a new type of backup integration type meant for non-AWS S3 compatible storage providers (e.g. MinIO, Cloudflare R2, Backblaze B2, ...). See https://github.com/stackrox/stackrox/pull/11877 and https://github.com/stackrox/stackrox/pull/11878 for the corresponding backend changes.

I have attempted to follow the patterns from the existing S3 and GCS back integrations types. Changelog and documentation will be done outside of this PR. I'm also planning a follow up to create a UI e2e test for these changes.

Config changes that are different from AWS S3 plugin:

* container IAM role is only supported for AWS S3 for now.
* S3 compatible plugin may configure a URL style setting.

SCR-20240708-mppb

SCR-20240708-msbd

### User-facing documentation

(*must be* 2 items and both *must be* checked)

- [ ] CHANGELOG is updated
- [x] CHANGELOG update is not needed
- [ ] Documentation PR is created and linked above
- [x] Documentation is not needed

### Testing

- [x] inspected CI results

#### Automated testing

(*must be* at least 1 item and all items *must be* checked)

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests
- [x] contributed **no automated tests**
  

#### How I validated my change

see screenshots

### CODE REVIEW

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/S3CompatibleIntegrationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+            }),
+            hour: yup.number(),
+            minute: yup.number(),
+        }),
+        s3compatible: yup.object().shape({
+            bucket: yup.string().trim().required('Bucket is required'),
+            objectPrefix: yup.string(),
+            endpoint: yup.string(),
+            region: yup.string().trim().required('Region is required'),
+            urlStyle: yup.string(),
```
Comment (lvalerom) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/S3CompatibleIntegrationForm.tsx:95]: Not sure if it makes sense to have this as `required`.

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/S3CompatibleIntegrationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+            secretAccessKey: yup
+                .string()
+                .trim()
+                .test(
+                    'secretAccessKey-test',
+                    'A secret access key is required',
+                    requireCredentials
+                ),
+        }),
+        type: yup.string().matches(/s3/),
```
Comment (lvalerom) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/S3CompatibleIntegrationForm.tsx:109]: `s3compatible` compatible?

**Review by lvalerom — COMMENTED** [review]
To the best of my knowledge this looks good to me. I've just added a few comments.

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/S3CompatibleIntegrationForm.tsx [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+export type S3CompatibleIntegration = {
+    s3compatible: {
+        bucket: string;
+        objectPrefix: string;
+        endpoint: string;
+        region: string;
+        accessKeyId: string;
+        secretAccessKey: string;
+        urlStyle: 'S3_URL_STYLE_PATH' | 'S3_URL_STYLE_VIRTUAL_HOSTED';
+    };
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/S3CompatibleIntegrationForm.tsx:52]: Although best to follow precedent of redundant declaration in most integrationforms, it is future tech debt task for UI team to import type from frontend proto (or even better, service) file as source of truth.

File: ui/apps/platform/src/Containers/Integrations/utils/tableColumnDescriptor.ts [hunk 4]
```diff
@@ -318,6 +318,21 @@ const tableColumnDescriptor: Readonly<IntegrationTableColumnDescriptorMap> = {
...
+            { accessor: 'name', Header: 'Name' },
+            { accessor: 's3compatible.bucket', Header: 'Bucket' },
+            {
+                accessor: ({ schedule }) => {
+                    if (schedule.intervalType === 'WEEKLY') {
+                        return `Weekly on ${daysOfWeek[schedule.weekly.day]} @ ${
+                            timesOfDay[schedule.hour]
+                        } UTC`;
+                    }
+                    return `Daily @ ${timesOfDay[schedule.hour]} UTC`;
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Integrations/utils/tableColumnDescriptor.ts:331]: Ouch, although better to be consistent, future design debt for UI team to confirm with tech docs that **at** is better than at symbol.

**Pull request merged**